src/lib.rs

```diff
@@ -118,6 +122,47 @@ impl<'a, V: VecScopedPrivate> VecScopedPrivate for Push<'a, V> {
 
 impl<'a, T, V: VecScopedPrivate<Element = T>> VecScoped<T> for Push<'a, V> {}
 
+pub struct Update<'a, V: VecScopedPrivate>(&'a mut V, usize, Option<V::Element>);
+
+impl<'a, V: VecScopedPrivate> Update<'a, V> {
+    pub fn new(vec_scoped: &'a mut V, value: V::Element, idx: usize) -> Self {
+        let inner = vec_scoped.vec_mut();
+        // i think this is efficient?
+        let old = inner.swap_remove(idx);
+        let end = inner.len();
+        inner.push(value);
+        inner.swap(end, idx);
+        Self(vec_scoped, idx, Some(old))
+    }
+}
+
+impl<'a, T, V: std::ops::Deref<Target = [T]> + VecScopedPrivate> std::ops::Deref for Update<'a, V> {
+    type Target = [T];
+
+    fn deref(&self) -> &Self::Target {
+        std::ops::Deref::deref(self.0)
```

Comment:
huh. 367f7d442fd71d83c347fc522564b30af58ff8d6 i'm  not sure i totally understand, but doesn't the `&*` cancel out? couldn't i just do `self.0`?

Reply:
Yeah, I had the same wonder. I think it's because `*` and `&` are not actually opposites: `*` applies `Deref` (which is necessary) but then you have a type `[T]` which is illegal so you need to take a reference to turn it into `&[T]`.

Reply:
but if i remove all the `&*`, everything still compiles and tests pass. what's the difference?

Reply:
Oh, that's confusing but good. In that case, could you make a PR removing the `&*`?